### PR TITLE
Feat: support user storage proxy for string type doc_id

### PR DESCRIPTION
**Pull request by Cologler**
for #221 .

examples see the added unittest.

### CODE REVIEW

File: tinydb/database.py [hunk 1]
```diff
@@ -77,6 +77,10 @@ def __init__(self, storage, table_name):
         self._storage = storage
         self._table_name = table_name
 
+    def construct_doc(self, key, val):
```
Comment (msiemens) [inline tinydb/database.py]: Could we rename this to an internal method and also use something like `def _new_document(…)`?

Comment (Cologler) [reply tinydb/database.py]: a new commit for that.

Comment (msiemens) [reply tinydb/database.py]: Awesome! Now you just need to adapt the example in https://github.com/msiemens/tinydb/blob/a454998e65e43f7af44e008129d08a44fcfed1e6/tests/test_tinydb.py#L617-L618 🙂

Comment (Cologler) [reply tinydb/database.py]: 😂

**Comment by msiemens** [comment]
Thanks for the idea, the contribution and the modifications, @Cologler! I'll try to bundle up a release of TinyDB next week 🙂

**Pull request merged**